glide-core/redis-rs/redis/src/cluster_routing.rs

```diff
@@ -264,7 +264,6 @@ pub(crate) fn combine_and_sort_array_results<'a>(
     for (key_indices, value) in sorting_order.into_iter().zip(values) {
         match value {
             Value::Array(values) => {
-                assert_eq!(values.len(), key_indices.len());
```

Comment:
This assert isn't correct, because `key_indices` contains argument indices, where not all args are keys. It also produces a useless message if fails:
```
thread '<unnamed>' panicked at glide-core/redis-rs/redis/src/cluster_routing.rs:269:17:
assertion `left == right` failed
  left: 1
 right: 2
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
2024-10-24T19:14:37.701195Z  WARN logger_core: received error - Received connection error `redis_cluster: Unable to receive command`. Will attempt to reconnect
```

As an option we can restore the assert, but make it producing a meaningful error and check an extra condition shipped in `ResponsePolicy::CombineArrays`

Reply:
`keys_indices` shouldn't contain argument indices, but the indices of the request in the original multi-request.
So if I have this MGET:
```
foo (idx 0)
bar (idx 1)
bar2 (idx 2)
bar4 (idx 3)
```

We will first check the route (e.g. slot) for each key, lets say we would get:
```
foo - slot 100
bar - slot 300
bar2 - slot 100
bar4 - slot 200
```
then we create a MultipleNodeRoutingInfo::MultiSlot with the following routes and indices from the request as follows:
```
[[(Route(slot(100)), indices: [0, 2]), (Route(slot(300)), indices: [1]), (Route(slot(200)), indices: [3])]
```

So the results of this MGET should have the same length: 
```
[[res_foo, res_bar2], [res_bar], [res_bar4]]
```

To summarize: key_indices contains the indices of the keys in the original requests and it should have the same length as the values in the results. As written in the description:
> `sorting_order` defines the order of the results in the returned array -
/// for each array of results, `sorting_order` should contain a matching array with the indices of
/// the results in the final array.


So if you get an error here it should be a real error we should address.
To make the assertion more readable you can add:

```
assert_eq!(
    values.len(), 
    key_indices.len(), 
    "Assertion failed: expected `values` and `key_indices` to have the same length, but got {} and {} respectively", 
    values.len(), 
    key_indices.len()
);